plaso/parsers/windefenderhistory.py

```diff
@@ -0,0 +1,319 @@
+    data_type_map = self._GetDataTypeMap('detection_history_value')
+
+    value, _ = self._ReadStructureFromFileObject(
+        file_object, file_offset, data_type_map)
+
+    value_object = None
+    if value.data_type in self._VALUE_DATA_TYPES_INTEGER:
+      value_object = value.value_integer
+    elif value.data_type == self._VALUE_DATA_TYPE_FILETIME:
+      value_object = value.value_filetime
+    elif value.data_type == self._VALUE_DATA_TYPE_STRING:
+      value_object = value.value_string
+    elif value.data_type == self._VALUE_DATA_TYPE_GUID:
+      value_object = value.value_guid
+    elif value.data_type == self._VALUE_DATA_TYPE_BINARY_DATA:
+      value_object = value.data
+    else:
+      parser_mediator.ProduceExtractionWarning(
+          'unknown value data type: {0!s}'.format(value.data_type))
+      value_object = value.data
+
+    return value_object
+
+  def ParseFileObject(self, parser_mediator, file_object):
+    """Parses a Windows Defender History file-like object.
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between
+            parsers and other components, such as storage and dfVFS.
+      file_object (dfvfs.FileIO): file-like object.
+
+    Raises:
+      WrongParser: when the file cannot be parsed.
+    """
+    value_tuples = []
+
+    file_offset = 0
+    file_size = file_object.get_size()
+
+    while file_offset < file_size:
+      value_object = self._ReadValue(file_object, file_offset, parser_mediator)
+      value_tuples.append((file_offset, value_object))
+      file_offset = file_object.tell()
+
+    resource_name = ''
+    threat_attributes = {'Resources': []}
+    value_index_set = 0
+    value_index = 0
+
+    for file_offset, value_object in value_tuples:
+      if value_object == self._FILE_SIGNATURE:
+        if value_index_set < 2:
+          value_index_set += 1
+        value_index = 0
+
+      if (value_index_set, value_index) == (2, 5):
+        threat_attributes.update(
+            self._ReadThreatTrackingData(value_object, file_offset + 8))
+
+      else:
+        description = self._VALUE_DESCRIPTIONS[value_index_set].get(
+            value_index, 'UNKNOWN_{0:d}_{1:d}'.format(
+                value_index_set, value_index))
+
+        value_string = '{0!s}'.format(value_object)
+
+        if description == 'Resource type':
+          resource_name = value_string
+
+        elif description == 'Resource location' and resource_name is not None:
+          threat_attributes['Resources'].append(
+              {'Type': resource_name, 'Location': value_string})
+          resource_name = None
+
+        else:
+          threat_attributes[description] = value_string
+
+      value_index += 1
+
+    filenames = [
+        threat_attribute['Location']
+        for threat_attribute in threat_attributes['Resources']
+        if threat_attribute['Type'] == 'file']
+
+    if not filenames:
+      filename = threat_attributes.get('CONTEXT_DATA_FILENAME', 'UNKNOWN')
```

Comment:
@Fryyyyy what is the idea/goal of setting these values to UNKNOWN? why not set them to None and not show what you don't know?

Reply:
Per chat with @Onager I'll change these to None seeing you'll likely be preoccupied at the moment.